flask_resty/view.py

```diff
@@ -415,20 +416,45 @@ def commit(self):
     def resolve_integrity_error(self, error):
         original_error = error.orig
 
-        if (
-            hasattr(original_error, 'pgcode') and
-            original_error.pgcode in (
+        if hasattr(original_error, 'pgcode'):
+            if original_error.pgcode in (
                 '23502',  # not_null_violation
                 '23514',  # check_violation
-            )
-        ):
-            # Using the psycopg2 error code, we can tell that this was not from
-            # an integrity error that was not a conflict. This means there was
-            # a schema bug, so we emit an interal server error instead.
-            return error
+            ):
+                # Using the psycopg2 error code, we can tell that this was not
+                # from an integrity error that was not a conflict. This means
+                # there was a schema bug, so we emit an internal server error
+                # instead.
+                return error
+            else:
+                diag = original_error.diag
+
+                schema_name = diag.schema_name
+                table_name = diag.table_name
+                constraint_name = diag.constraint_name
+
+                insp = sa.inspect(self.session.bind)
+                unique_constraints = insp.get_unique_constraints(
+                    table_name,
+                    schema=schema_name,
+                )
+
+                for unique_constraint in unique_constraints:
+                    if unique_constraint['name'] == constraint_name:
+                        column_names = unique_constraint['column_names']
+                        break
+                else:
+                    column_names = ()
+        else:
+            column_names = ()
 
         flask.current_app.logger.exception("handled integrity error")
-        return ApiError(409, {'code': 'invalid_data.conflict'})
+
+        error = {'code': 'invalid_data.conflict'}
+        if column_names:
+            error['source'] = {'pointer': '/data/{}'.format(column_names[-1])}
```

Comment:
For multi-column unique constraints (e.g. tuple of `tenant_id`, `name`), we generally want to report the conflict on the last column, but this is something we need to tweak.

Reply:
I think that assumption would require some level of documentation. Alternatively could make the default to show all columns and then have a hook in the view to allow the view to reduce what is shown.

Reply:
docs for sure. the issue is that we have a lot of unique constraints like `(tenant_id, name)`, and it would be odd to show an error on both fields in such cases

Reply:
we could also have a `pointers` field being an array of pointers. we are already not jsonapi compliant anyway :trollface: